tests/rptest/utils/functional.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from functools import reduce
+
+
+def flat_map(fn, input_list, initial_value=[]):
+    """
+    Expect a function that maps an argument to a list of arguments.
```

Comment:
nit: I found this a bit confusing since the outputs aren't really "arguments". Maybe reword as "Expect a function that maps an argument to a List type result." or something?

Reply:
the function i'm talking about is `fn`. I can expand on that though